src/main/java/interface_adapters/tui/controllers/RecipeNoteOperation.java

```diff
@@ -0,0 +1,159 @@
+package interface_adapters.tui.controllers;
+
+import entity.Note;
+import entity.Recipe;
+import interface_adapters.tui.Colour;
+import interface_adapters.tui.TextualOperation;
+import interface_adapters.tui.TextualReader;
+import usecase.RecipeManager;
+import usecase.RecipeNoteTaker;
+
+import java.util.List;
+
+
+/**
+ * An operation for running the note taker use-case.
+ */
+public class RecipeNoteOperation implements TextualOperation {
+    private final TextualReader reader;
+    private final RecipeNoteTaker recipeNoteTaker;
+    private final RecipeManager recipeManager;
+
+    private final List<TextualOperation> operations = List.of(new NoteCreator(), new NoteViewer(), new NoteEditor(), new NoteDeleter());
+
+    public RecipeNoteOperation(TextualReader reader, RecipeNoteTaker recipeNoteTaker, RecipeManager recipeManager) {
+        this.reader = reader;
+        this.recipeNoteTaker = recipeNoteTaker;
+        this.recipeManager = recipeManager;
+    }
+
+    @Override
+    public void run() {
+        reader.chooseOperation(operations, "Recipe Note Taker");
+    }
+
+    @Override
+    public String getDescription() {
+        return "Create, list, edit, and delete note";
+    }
+
+    @Override
+    public String getCode() {
+        return "recipe note taker";
+    }
+
+    private class NoteCreator implements TextualOperation {
+
+        @Override
+        public String getCode() {
+            return "create note";
+        }
+
+        @Override
+        public String getDescription() {
+            return "Create a new note";
+        }
+
+        @Override
+        public void run() {
+            String title = reader.getInput("Recipe Title:");
+            Integer servings = reader.getIntegerInput("Servings (e.g. 10):");
+            Integer prep_time = reader.getIntegerInput("Minutes of prep time:");
```

Comment:
It's not clear to me why the note taker should be making new recipes. As it is specified, it should only add notes to already existing recipes.